influent-java/src/main/java/influent/forward/MsgPackPingDecoder.java

```diff
@@ -0,0 +1,122 @@
+package influent.forward;
+
+import org.msgpack.value.ImmutableArrayValue;
+import org.msgpack.value.ImmutableValue;
+import org.msgpack.value.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Optional;
+
+final class MsgPackPingDecoder {
```

Comment:
This class has the responsibility to test `PING` request.
So `decode` may not be suitable.

Reply:
How about `PingMessage`  and `PingMessage#check`?